src/shared/common/forms/form-repository.jsx

```diff
@@ -159,6 +160,12 @@ export function formComponentFactory(field, options) {
             return renderFormComponent(Inset, { key, data, config });
         case 'confirmation-with-case-ref':
             return renderFormComponent(ConfirmationWithCaseRef, { key, data, config, caseRef });
+        case 'confirmation-with-team-name-and-case-ref':
+            var teamName;
+            if (data.AcceptanceTeam){
```

Comment:
Can this logic be moved into the component? It will keep the switch a little cleaner. I know there are already some instances where someone has added logic in the case block but really it's polluting the class a little and you seem to be passing all that's needed to the component anyway.

Reply:
Ah, the reason I kept this outside was due to potential to extend this with other team types, the team name may not always be passed through an AcceptanceTeam variable, so I figured this was the best way to keep it usable in other functionalities. Can still change it if you want.

Reply:
Hi @LaurieBennett , I still think this logic should be moved into the component for the same reason state above. I also think that rather than passing the `AcceptanceTeam` variable into the component, extract the team from `data` once inside the component. That follows the pattern of the method. 

In the component, itself the `AcceptanceTeam` should be extracted. I accept your point that this component should be usable by teams other than `AcceptanceTeams` so a solution will need to be worked out. A potential solution is to filter the `data` object for the type of team required based on the `CurrentStage`, and use a key-value pair to say... if current stage is X then I'm looking for a team type of Y. Give me a shout if you want any help.